node/src/components/block_executor.rs

```diff
@@ -417,9 +430,18 @@ impl BlockExecutor {
         state_root_hash: Digest,
         next_era_validator_weights: Option<BTreeMap<PublicKey, U512>>,
     ) -> Block {
-        let (parent_summary_hash, parent_seed) = if finalized_block.is_genesis_child() {
-            // Genesis, no parent summary.
-            (BlockHash::new(Digest::default()), Digest::default())
+        let (parent_summary_hash, parent_seed) = if self.is_initial_block_child(&finalized_block) {
+            // The first block after the initial one, no parent summary.
```

Comment:
We won't have one for the initial block, either, because we won't even try to execute any older blocks.
And even if we had, we shouldn't be looking at it, because it might contain wrong data.

Reply:
But in the next lines, we _are_ taking `self.initial_block_summary`, which we _did_ initialize in the constructor?

Reply:
I'm just saying this comment used to make sense when in the next line it said `default()`, but now it doesn't fully apply anymore.